tests/integration/vpp/080_gre_test.go

```diff
@@ -0,0 +1,127 @@
+package vpp
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ligato/cn-infra/logging/logrus"
+	interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	ifplugin_vppcalls "github.com/ligato/vpp-agent/plugins/vpp/ifplugin/vppcalls"
+)
+
+func TestGre(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	h := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+
+	tests := []struct {
+		name    string
+		greLink *interfaces.GreLink
+		isFail  bool
+	}{
+		{
+			name: "create ERSPAN GRE tunnel with IPv4",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "10.10.10.10",
+				DstAddr:    "20.20.20.20",
+			},
+			isFail: false,
+		},
+		{
+			name: "create ERSPAN GRE tunnel with out of range session id",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "10.10.10.10",
+				DstAddr:    "20.20.20.20",
+				SessionID:  1024,
+			},
+			isFail: true,
+		},
+		{
+			name: "create GRE tunnel with bad source address",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "badip",
+				DstAddr:    "20.20.20.20",
+			},
+			isFail: true,
+		},
+		{
+			name: "create GRE tunnel with destination address not set",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "10.10.10.10",
+			},
+			isFail: true,
+		},
+
+		{
+			name: "create GRE tunnel with equal source and destination addresses",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "10.10.10.10",
+				DstAddr:    "10.10.10.10",
+			},
+			isFail: true,
+		},
+		{
+			name: "create GRE tunnel with addresses in IPv4 and IPv6",
+			greLink: &interfaces.GreLink{
+				TunnelType: interfaces.GreLink_ERSPAN,
+				SrcAddr:    "10.10.10.10",
+				DstAddr:    "2019::08:23",
+			},
+			isFail: true,
+		},
+	}
+	for i, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ifIdx, err := h.AddGreTunnel(fmt.Sprintf("test%d", i), test.greLink)
+
+			if err != nil {
```

Comment:
I'm not sure this will handle test case where we expect some non-nil error, but it gets `nil`.